src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        switch (random.nextInt(5)) {
```

Comment:
All magic numbers in your code should be constants. Instead of using '5' directly, you should define it as a constant with a clear name that indicates its purpose. For example, `private static final int FIGURE_COUNT = 5;` and then use `FIGURE_COUNT` in the `nextInt` call.